src/components/themed/WalletListCreateRow.js

```diff
@@ -1,185 +1,131 @@
+        const tokens = enabledTokens.filter(tokenId => tokenId !== wallet?.currencyInfo?.pluginId)
+        await setEnabledTokens(wallet, [...tokens, currencyCode], [])
+        return [...enabledTokens, currencyCode]
       }
 
-      const guiWalletEnabledTokens = wallets[wallet.id]?.enabledTokens ?? []
-      const enabledTokens = await showFullScreenSpinner(
-        s.strings.wallet_list_modal_enabling_token,
-        // Should use EdgeCurrencyWallet.getEnabledTokens() but function seems to return parent currency code as part of the array
-        setEnabledTokens(wallet, [...guiWalletEnabledTokens, currencyCode], [])
-      )
+      const enabledTokens = await showFullScreenSpinner(s.strings.wallet_list_modal_enabling_token, addToken())
+
+      dispatch({
+        type: 'UPDATE_WALLET_ENABLED_TOKENS',
+        data: { walletId, tokens: enabledTokens }
+      })
+      dispatch(refreshWallet(walletId))
 
-      tokenCreated(wallet.id, enabledTokens)
-      onPress(wallet.id, currencyCode)
+      return walletId
     } catch (error) {
       showError(error)
     }
+    return ''
   }
 
-  handlePress = () => {
-    if (this.props.createWalletType) {
-      this.createAndSelectWallet().catch(showError)
-    } else {
-      this.createAndSelectToken().catch(showError)
+export const createAndSelectWallet = ({ walletType, fiatCurrencyCode, walletName }: CreateWalletOptions) => {
+  walletName = walletName ?? getSpecialCurrencyInfo(walletType).initWalletName
+  return async (dispatch: Dispatch, getState: GetState) => {
+    const state = getState()
+    const { account } = state.core
+    try {
+      const wallet = await showFullScreenSpinner(
+        s.strings.wallet_list_modal_creating_wallet,
+        createWallet(account, { walletName, walletType, fiatCurrencyCode })
+      )
+      return wallet.id
+    } catch (error) {
+      showError(error)
     }
+    return ''
   }
+}
 
-  renderIcon() {
-    const { createWalletType, createTokenType } = this.props
-    const styles = getStyles(this.props.theme)
-    return (
-      <View style={styles.iconContainer}>
-        <FastImage style={styles.iconSize} source={{ uri: createWalletType?.symbolImage ?? createTokenType?.symbolImage ?? '' }} />
-      </View>
-    )
-  }
+export const WalletListCreateRowComponent = (props: WalletListCreateRowProps) => {
+  const dispatch = useDispatch()
+  const theme = useTheme()
+  const styles = getStyles(theme)
+  const { currencyCode = '', currencyName = '', walletType, symbolImage = '', symbolImageDarkMono = '', parentCurrencyCode, onPress } = props
+
+  const handlePress = useCallback(() => {
+    const handleRes = walletId => (onPress != null ? onPress(walletId, currencyCode) : null)
+    if (walletType != null) dispatch(createAndSelectWallet({ walletType })).then(handleRes)
+    if (parentCurrencyCode != null) dispatch(createAndSelectToken({ currencyCode, parentCurrencyCode })).then(handleRes)
```

Comment:
The two `if` statements should be mutually exclusive, so we need an `else` here. Otherwise, `onPress` might be called twice.